modules/ml/src/main/java/org/apache/ignite/ml/nn/architecture/MLPArchitecture.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.nn.architecture;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.ml.math.functions.IgniteDifferentiableDoubleToDoubleFunction;
+
+/**
+ * Class containing information about architecture of MLP.
+ */
+public class MLPArchitecture {
+    /**
+     * List of layers architectures.
+     */
+    private final List<LayerArchitecture> layers;
+
+    /**
+     * Construct an MLP architecture.
+     *
+     * @param inputSize Size of input.
+     */
+    public MLPArchitecture(int inputSize) {
+        layers = new ArrayList<>();
+        layers.add(new LayerArchitecture(inputSize));
+    }
+
+    /**
+     * Construct an MLP architecture.
+     *
+     * @param layers List of layers architectures.
+     */
+    private MLPArchitecture(List<LayerArchitecture> layers) {
+        this.layers = layers;
+    }
+
+    /**
+     * Count of layers in MLP.
+     * @return Layers count.
+     */
+    public int layersCount() {
+        return layers.size();
+    }
+
+    /**
+     * Size of input of MLP.
+     * @return Size of input.
+     */
+    public int inputSize() {
+        return layers.get(0).neuronsCount();
+    }
+
+    /**
+     * Size of output of MLP.
+     * @return Size of output.
+     */
+    public int outputSize() {
+        return layers.get(layersCount() - 1).neuronsCount();
+    }
+
+    /**
+     * Constructs new MLP architecture with new layer added on top of all this architecture layers.
+     *
+     * @param neuronsCnt Count of neurons in new layer.
+     * @param hasBias Flag indicating presence of bias in added layer.
+     * @param f Activation function of a new layer.
+     * @return New MLP architecture with new layer added on top of all this architecture layers.
+     */
+    public MLPArchitecture withAddedLayer(int neuronsCnt, boolean hasBias, IgniteDifferentiableDoubleToDoubleFunction f) {
+        ArrayList<LayerArchitecture> newLayers = new ArrayList<>(layers);
+        newLayers.add(new TransformationLayerArchitecture(neuronsCnt, hasBias, f));
+        return new MLPArchitecture(newLayers);
+    }
+
+    /**
+     * Get architecture of layer with given index.
+     *
+     * @param layer Index of layer to get architecture from.
+     * @return Architecture of layer with given index.
+     */
+    public LayerArchitecture layerArchitecture(int layer) {
+        return layers.get(layer);
+    }
+
+    /**
+     * Get architecture of transformation layer (i.e. non-input layer) with given index.
+     *
+     * @param layer Index of layer to get architecture from.
+     * @return Architecture of transformation layer with given index.
+     */
+    public TransformationLayerArchitecture transformationLayerArchitecture(int layer) {
```

Comment:
What's the reason to return specific type instead abstract

Reply:
To avoid casting. TransformationLayerArchitecture contain more information than just LayerArchitecture.